test-data/unit/check-optional.test

```diff
@@ -105,9 +105,12 @@ f(None)
 [out]
 main: note: In function "f":
 
-[case testInferOptionalType]
+[case testInferOptionalTypeWithBinder]
 x = None
-x = 1
+if 0:
```

Comment:
I like to use `if bool():` for tests like these (and converted a bunch of tests to that style in the binder PR).